tests/factories.py

```diff
@@ -115,3 +115,47 @@ class Meta:
 class NominatedURLFactory(URLFactory):
     attribute = 'nomination'
     value = fuzzy.FuzzyChoice(['-1', '1'])
+
+
+class SURTFactory(URLFactory):
+    attribute = 'surt'
+    value = fuzzy.FuzzyText(length=5, prefix='http://(com,', suffix=',www,)')
+
+
+class MetadataWithValuesFactory(MetadataFactory):
+    value1 = factory.RelatedFactory(MetadataValuesFactory, 'metadata')
+    value2 = factory.RelatedFactory(MetadataValuesFactory, 'metadata')
+    value3 = factory.RelatedFactory(MetadataValuesFactory, 'metadata')
+
+
+class ValuesetWithValuesFactory(ValuesetFactory):
+    value1 = factory.RelatedFactory(ValuesetValuesFactory, 'valueset')
+    value2 = factory.RelatedFactory(ValuesetValuesFactory, 'valueset')
+    value3 = factory.RelatedFactory(ValuesetValuesFactory, 'valueset')
+
+
+class MetadataWithValuesetFactory(MetadataFactory):
+    @factory.post_generation
+    def value_sets(self, create, extracted, **kwargs):
+        if not create:
+            return
+
+        elif extracted:
+            for value_set in extracted:
+                self.value_sets.add(value_set)
+
+        else:
+            self.value_sets.add(ValuesetWithValuesFactory())
+
+
+class ProjectWithMetadataFactory(ProjectFactory):
```

Comment:
I think I'm going to stick with the way it is. I tried a couple ways and they didn't seem to be working. Also, this is actually exactly how the docs say to do something like this, where you want factories for a model that has a many-to-many relationship with a "through".